sound/soc/sof/pm.c

```diff
@@ -156,15 +156,6 @@ static int sof_resume(struct device *dev, bool runtime_resume)
 			 ret);
 	}
 
-	/* restore pipelines */
```

Comment:
@keyonjie there was a specific reason we wanted to keep dynamic pipelines as an option. There will be cases such as the DSM pipelines that will need to be static and cannot be dynamic

Reply:
dynamic pipelines will add a mechanical latency to restore the information before you can use a given pipeline. It might be useful in some contexts to do the pipeline setup once, e.g. a critical sound is played without delays. 

It's not science-fiction, in the past we were even asked to embed the topology information in tables so that we could shave all the IPC time required on startup.

Reply:
get  your point @ranj063 @plbossart , I would suggest to make 'dynamic' as the default setting for pipelines, and specify 'static' token in the topology to ask for being resident ones, I just check the logs, setting up a 5~8 widgets traditional pipeline is quite 'fast' -- takes about 1~2 ms, comparing with the DSP suspend/resume (and FW re-downloading) which takes 40~50ms on APL UP2.

Reply:
@keyonjie we want to proceed gradually and start migrating topologies by groups from static to dynamic. We can't take the risk of breaking all configurations at once. @ranj063 already added the HDaudio-based topologies, I believe SoundWire is next. I don't think we want to touch anything used for I2S platforms prior to TGL. there's no real point in making dynamic pipelines the default across the board, including for legacy or already shipping devices.

Reply:
> @keyonjie we want to proceed gradually and start migrating topologies by groups from static to dynamic. We can't take the risk of breaking all configurations at once. @ranj063 already added the HDaudio-based topologies, I believe SoundWire is next. I don't think we want to touch anything used for I2S platforms prior to TGL. there's no real point in making dynamic pipelines the default across the board, including for legacy or already shipping devices.

@plbossart get your point. It's up to you maintainers to decide how to proceed, I am only sharing my 2 cents with different suggestion, if we have enough confidence about what we are changing and believe it is the better solution, we should provide these updates even to the shipping devices after enough validation. Making dynamic as default looks natural and will reduce our maintaining effort IMHO.